solarforecastarbiter/io/api.py

```diff
@@ -281,17 +285,39 @@ def create_forecast(self, forecast):
         -------
         datamodel.Forecast
             With the appropriate parameters such as forecast_id set by the API
+
         """
         fx_dict = forecast.to_dict()
         fx_dict.pop('forecast_id')
         site = fx_dict.pop('site')
-        fx_dict['site_id'] = site['site_id']
+        agg = fx_dict.pop('aggregate')
+        if site is None and agg is not None:
+            fx_dict['aggregate_id'] = agg['aggregate_id']
+        else:
+            fx_dict['site_id'] = site['site_id']
```

Comment:
should we raise exception `if site is not None and agg is not None`? Probably no because I don't think it would be possible to hit that line in this method since you're starting with a valid Forecast object.

Reply:
exactly. If somehow both are None, this would raise an error anyway.